### PR TITLE
play better with coc.nvim, set hl-mode to 'combine'

### PR DESCRIPTION
**Pull request by daquexian**
Thanks for your great plugin!

I'm using copilot.vim in together with coc.nvim, the most popular lsp client for nvim. But the experience is not good: ghost text disappears when the pmenu is shown, flickers when typing, and conflict with nvim background color.

I made some changes in copilot.vim to get better experience. Please see the below screen recordings for comparison.

New behavior (plays well with pmenu and background color, ghost texts don't flicker anymore):

https://user-images.githubusercontent.com/11607199/163096904-98236b5b-cf22-462a-8ee6-0b37b04c159b.mov

Old behavior (conflicts with pmenu and background color, ghost texts flicker when typing):

https://user-images.githubusercontent.com/11607199/163098219-645936ec-2649-45b3-844d-cb8d11477f02.mov

### CODE REVIEW

File: autoload/copilot.vim [hunk 1]
```diff
@@ -171,7 +171,7 @@ function! copilot#NvimNs() abort
   return nvim_create_namespace('github-copilot')
 endfunction
 
-function! copilot#Clear() abort
```
Comment (daquexian) [inline autoload/copilot.vim:174]: The `Clear` function doesn't unlet `b:_copilot` anymore, and is separated into two functions `CancelOngoingRequest` and `UpdatePreview`.

File: autoload/copilot.vim [hunk 2]
```diff
@@ -257,7 +256,7 @@ endfunction
 
 function! s:SuggestionTextWithAdjustments() abort
   try
-    if mode() !~# '^[iR]' || pumvisible() || !s:dest || !exists('b:_copilot.suggestions')
+    if mode() !~# '^[iR]' || !s:dest || !exists('b:_copilot.suggestions')
```
Comment (daquexian) [inline autoload/copilot.vim:259]: show ghost text even if `pumvisible()` is true

File: autoload/copilot.vim [hunk 3]
```diff
@@ -430,7 +429,7 @@ function! s:UpdatePreview() abort
     else
       let data.virt_text += annot
     endif
-    call nvim_buf_del_extmark(0, copilot#NvimNs(), 1)
+    let data.hl_mode = 'combine'
```
Comment (daquexian) [inline autoload/copilot.vim:432]: no need to call `nvim_buf_del_extmark` -- `nvim_buf_set_extmark` will overrides the old one

File: autoload/copilot.vim [hunk 4]
```diff
@@ -470,7 +469,8 @@ endfunction
 let s:is_mapped = copilot#IsMapped()
 
 function! copilot#Schedule(...) abort
-  call copilot#Clear()
+  call s:UpdatePreview()
+  call copilot#CancelOngoingRequest()
```
Comment (daquexian) [inline autoload/copilot.vim:473]: `b:_copilot` is not "unlet-ed" now

File: autoload/copilot.vim [hunk 5]
```diff
@@ -493,7 +492,7 @@ function! copilot#OnInsertEnter() abort
 endfunction
 
 function! copilot#OnCompleteChanged() abort
-  return copilot#Clear()
+  return copilot#Schedule()
```
Comment (daquexian) [inline autoload/copilot.vim:495]: schedule instead of clear when complete changed

**Comment by tpope** [comment]
Thanks for the feedback, I'll take it under consideration. The automatic hiding was designed to be paired with manual complete.

File: autoload/copilot.vim (hunk 3 above)
Comment (tpope) [reply autoload/copilot.vim:432]: This works around a bug in some versions of Neovim.

**Comment by daquexian** [comment]
> The automatic hiding was designed to be paired with manual complete.

I see. It will be appreciated if you provide a configuration flag for auto-complete plugin users :)

**Pull request closed without merging**